cmd/argo-watcher/argo_status_updater.go

```diff
@@ -76,6 +78,20 @@ func (updater *ArgoStatusUpdater) waitForApplicationDeployment(task models.Task)
 	var application *models.Application
 	var err error
 
+	app, err := updater.argo.api.GetApplication(task.App)
+	if err != nil {
+		return nil, err
+	}
+
+	if app.IsManagedByWatcher() && task.Validated {
```

Comment:
I think the token verification should happen in the router. In case the token is invalid, fail the API call. In case it's successful - continue the execution. And for git commit deployment identification in the code, I'd add a separate parameter, for example `GitCommitEnabled`, so that it's explicit what the client wants.